packages/server/src/adapters/aws-lambda/content-type/json/index.ts

```diff
@@ -0,0 +1,99 @@
+// @trpc/server
+import { TRPCError } from '../../../../@trpc/server';
+import type {
+  AnyRouter,
+  CombinedDataTransformer,
+} from '../../../../@trpc/server';
+import type {
+  BaseContentTypeHandler,
+  HTTPRequest,
+} from '../../../../@trpc/server/http';
+import {
+  lambdaEventToHTTPBody,
+  type APIGatewayEvent,
+  type AWSLambdaOptions,
+} from '../../utils';
+
+export interface LambdaHTTPContentTypeHandler<
+  TRouter extends AnyRouter,
+  TEvent extends APIGatewayEvent,
+> extends BaseContentTypeHandler<
+    AWSLambdaOptions<TRouter, TEvent> & {
+      event: TEvent;
+      req: HTTPRequest;
+    }
+  > {}
+
+export const getLambdaHTTPJSONContentTypeHandler: <
+  TRouter extends AnyRouter,
+  TEvent extends APIGatewayEvent,
+>() => LambdaHTTPContentTypeHandler<TRouter, TEvent> = () => ({
+  name: 'lambda-json',
+  isMatch(opts) {
+    return !!opts.event.headers['Content-Type']?.startsWith('application/json');
+  },
+  getInputs: async (opts, info) => {
+    function getRawProcedureInputOrThrow() {
+      const { event, req } = opts;
+
+      try {
+        if (req.method === 'GET') {
+          const input = req.query.get('input');
+          if (!input) {
+            return undefined;
+          }
+
+          return JSON.parse(input);
+        }
+
+        const body = lambdaEventToHTTPBody(opts.event);
+        if (typeof body === 'string') {
+          // A mutation with no inputs will have req.body === ''
+          return body.length === 0 ? undefined : JSON.parse(body);
+        }
+        return event.body;
+      } catch (cause) {
+        throw new TRPCError({
+          code: 'PARSE_ERROR',
```

Comment:
Nice

Reply:
Ha, I didn't even write this, again it's a C&P job when refactoring. You may have just complimented yourself :D